feat: add position sharing functionality and mark price display OK-45182

Minnzen opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Position sharing: per-position Share button, customizable share cards (backgrounds, stickers, custom text), toggle PnL/ROE, save image, copy link, or post to X/Twitter; responsive modal and mobile flow.

* **Improvements**
  * Increased PnL column width for better readability in positions list.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Minnzen committed:
feat: add position sharing functionality and mark price display

- Introduced a new share feature in the PositionRow component, allowing users to share position details.
- Added a MarkPriceFromContext component to display the mark price in the PositionRow.
- Updated routing to include a PositionShare modal.
- Enhanced type definitions to support sharing data structure.

Kahnchan committed:
Feat/perps share (#8922)

* refactor: update PositionShare styles and layout for improved visuals

- Changed color values for long and short positions to enhance visibility.
- Adjusted font sizes and layout dimensions for better alignment and aesthetics.
- Introduced a referral background color for added context in the share image.
- Updated text positioning and baseline settings for improved text rendering in the share image generator.

* feat: enhance PositionShare component with new background colors and layout adjustments

- Added new background colors for long and short positions to improve visual distinction.
- Removed the customizable text input from the ControlPanel for a cleaner interface.
- Updated ShareContentRenderer to use absolute positioning for better layout control.
- Enhanced the ShareImageGenerator to utilize the new color constants for background rendering.
- Adjusted type definitions to include new color properties for consistency.

* feat: update PositionShare layout and styling for enhanced user experience

- Adjusted referral background color for improved visibility.
- Refined layout properties including badge padding, spacing, and line height for better alignment.
- Updated ShareContentRenderer and ShareImageGenerator to utilize new layout configurations for consistent styling.
- Enhanced type definitions to reflect new layout attributes.

* feat: refine PositionShare component layout and scaling for improved responsiveness

- Introduced a scaling function to adjust layout dimensions and font sizes based on the current screen size.
- Updated canvas configuration to use dynamic sizing for better adaptability across devices.
- Adjusted modal dimensions for a more consistent user experience on different screen sizes.

* feat: enhance PositionShare component with dynamic canvas configuration and layout improvements

- Refactored canvas configuration to allow dynamic sizing based on the current screen size.
- Updated ControlPanel and ShareContent components for improved layout and responsiveness.
- Adjusted styling and spacing for better visual consistency across different devices.
- Enhanced ShareImageGenerator to utilize the new canvas configuration for optimized image rendering.

* i18n

* refactor: improve layout and styling in OrderInfoPanel and PositionShare components

- Adjusted spacing and gap values in PositionsRow for better alignment.
- Updated icon sizes and added share functionality in mobile header.
- Increased minimum width for PerpPositionsList for improved readability.
- Refined ControlPanel layout with flex properties for better responsiveness.
- Enhanced ShareContent layout with consistent spacing and dimensions for a cleaner appearance.

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/bff7da40-57f2-465f-a24e-767309a042a7?refs=92c0f26a9f8e036dc7789ccf448a072f349d5687&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/bff7da40-57f2-465f-a24e-767309a042a7/license?refs=92c0f26a9f8e036dc7789ccf448a072f349d5687&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

originalix commented on packages/kit/src/views/Perp/components/PositionShare/ControlPanel.tsx:
```diff
@@ -0,0 +1,154 @@
+
+  const isProfit = useMemo(() => {
+    const pnlNum = parseFloat(data.pnl);
+    return pnlNum >= 0;
```
bigNumber.isGreaterThan(0)

originalix commented on packages/kit/src/views/Perp/components/PositionShare/ShareContentRenderer.tsx:
```diff
@@ -0,0 +1,294 @@
+    markPrice,
+  } = data;
+  const pnlNum = parseFloat(pnl);
+  const pnlPercentNum = parseFloat(pnlPercent);
```
BN

originalix commented on packages/kit/src/views/Perp/components/PositionShare/useShareActions.ts:
```diff
@@ -0,0 +1,105 @@
+  const shareToX = useCallback(async (base64Image: string, text: string) => {
+    try {
+      const tweetText = `${text}\n\n${REFERRAL_CODE}`;
+      const twitterUrl = `https://twitter.com/intent/tweet?text=${encodeURIComponent(
```
x.com

Minnzen committed:
feat: enhance PositionShare component with PNL display mode and background handling

- Introduced a new PNL display mode feature, allowing users to toggle between 'roe' and 'pnl' views.
- Refactored background handling to dynamically generate URLs for background images based on a CDN.
- Updated ControlPanel to manage PNL display mode selection and removed unused input fields for a cleaner interface.
- Enhanced ShareContentRenderer and ShareImageGenerator to utilize the new PNL display information and background configurations.
- Adjusted type definitions to include the new PNL display mode property for consistency.

Minnzen committed:
fix: lint
